### ISSUE TITLE
Fix install editor section of setup

### ISSUE THREAD

**Issue by ZiyadF296** [FlutterMatic/desktop#33]
**Describe the bug**

**To Reproduce**
- Start the setup
- Go to install editors section
- Select both editors (VSC and Studio)
- Click check

**Expected behavior**
It is expected to stop loading once the work is done. Proven to be done by checking the logs.

**Desktop (please complete the following information):**
 - OS: Windows (I am confident it's all platforms)

**Comment by ZiyadF296** [comment]
@yahu1031 Check this issue and work on it if you can.
Keep my updated.

**Comment by yahu1031** [comment]
The API is up. But need some time to avoid usage of this API. I'm trying to avoid API calls due to frequent updates of API data is required.

**Comment by ZiyadF296** [comment]
> The API is up. But need some time to avoid usage of this API. I'm trying to avoid API calls due to frequent updates of API data is required. This can be removed as `p1` and Just add a `backlog` tag/label.

This issue is a different issue and I filed a new issue for it. It can be worked on later after the setup flow is smooth and polished. The install editor part glitches because of `provider` not updating `progress` correctly and appropriately.

**Comment by ZiyadF296** [comment]
This issue has been resolved and changes will be seen in the next push.
